src/Compilers/CSharp/Portable/Binder/LookupResult.cs

```diff
@@ -191,7 +191,13 @@ internal static SingleLookupResult NotTypeOrNamespace(Symbol symbol, DiagnosticI
         internal static SingleLookupResult NotTypeOrNamespace(Symbol unwrappedSymbol, Symbol symbol, bool diagnose)
         {
             // TODO: determine correct diagnosis 
-            var diagInfo = diagnose ? new CSDiagnosticInfo(ErrorCode.ERR_BadSKknown, unwrappedSymbol, unwrappedSymbol.GetKindText(), MessageID.IDS_SK_TYPE.Localize()) : null;
+            object sym = unwrappedSymbol;
+            // for CS0118 we dont want to display the type, but the name
+            if (unwrappedSymbol.Kind == SymbolKind.Parameter && diagnose)
```

Comment:
This is a one-off patch for a symptom. Can we fix the root of the problem by overriding `ToString` in `ParameterSymbol.cs`?